sdk/nodejs/provider/server.ts

```diff
@@ -62,10 +62,13 @@ class Server implements grpc.UntypedServiceImplementation {
     }
 
     public getSchema(call: any, callback: any): void {
-        callback({
-            code: grpc.status.UNIMPLEMENTED,
-            details: "Not yet implemented: GetSchema",
-        }, undefined);
+        const req: any = call.request;
+        if (req.getVersion() !== 0) {
+            callback(new Error(`unsupported schema version ${req.getVersion()}`), undefined);
```

Comment:
Do `Error`s bubble up correctly here?  Or is returning `UNIMPLEMENTED` in this case going to behave better?

Reply:
> Do `Error`s bubble up correctly here?

Yes, from a client's perspective this returns the same error as providers written in other languages and is the same error our providers return (e.g. [k8s](https://github.com/pulumi/pulumi-kubernetes/blob/0072954b2cdf088fc2e336ca4c289929f75ec1a5/provider/pkg/provider/provider.go#L188-L190), [tfbridge](https://github.com/pulumi/pulumi-terraform-bridge/blob/30ae8ae5cc2d2f5987854984243c74ace4ccc952/pkg/tfbridge/provider.go#L294-L296)).